h2o-core/src/main/java/water/init/NetworkInit.java

```diff
@@ -185,6 +197,59 @@ public static void initializeNetworkSockets( ) {
     H2O.CLOUD_MULTICAST_PORT = NetworkUtils.getMulticastPort(hash);
   }
 
+  public static H2OHttpConfig webServerParams(H2O.OptArgs args) {
+    final H2OHttpConfig params = new H2OHttpConfig();
+    params.jks = args.jks;
+    params.jks_pass = args.jks_pass;
+    params.loginType = parseLoginType(args);
+    params.login_conf = args.login_conf;
+    params.form_auth = args.form_auth;
+    params.session_timeout = args.session_timeout;
+    params.user_name = args.user_name;
+    params.port = args.port;
+    params.baseport = args.baseport;
+    params.web_ip = args.web_ip;
+    params.ip = args.ip;
+    params.network = args.network;
+    params.context_path = args.context_path;
+
+    params.h2oExtendedHeaders = new LinkedHashMap<>();
+    params.h2oExtendedHeaders.put("X-h2o-build-project-version", H2O.ABV.projectVersion());
+    params.h2oExtendedHeaders.put("X-h2o-rest-api-version-max", Integer.toString(water.api.RequestServer.H2O_REST_API_VERSION));
+    params.h2oExtendedHeaders.put("X-h2o-cluster-id", Long.toString(H2O.CLUSTER_ID));
+    params.h2oExtendedHeaders.put("X-h2o-cluster-good", Boolean.toString(H2O.CLOUD.healthy()));
+    params.h2oExtendedHeaders.put("X-h2o-context-path", ServletUtils.sanatizeContextPath(H2O.ARGS.context_path));
+
+    return params;
+  }
+
+  private static LoginType parseLoginType(H2O.BaseArgs args) {
+    final LoginType loginType;
+    if (args.hash_login) {
+      loginType = LoginType.HASH;
+    } else if (args.ldap_login) {
+      loginType = LoginType.LDAP;
+    } else if (args.kerberos_login) {
+      loginType = LoginType.KERBEROS;
+    } else if (args.pam_login) {
+      loginType = LoginType.PAM;
+    } else {
+      return LoginType.NONE;
+    }
+    if (args.login_conf == null) {
+      throw new IllegalArgumentException("Must specify -login_conf argument");
+    }
+    if (loginType.isJaas()) {
+      // LDAP, KERBEROS, PAM
+      Log.info(String.format("Configuring JAASLoginService (with %s)", loginType));
+      System.setProperty("java.security.auth.login.config", args.login_conf);
```

Comment:
This method is called parseLoginType but it has a side effect. Consider changing the method so that operates without a side effect.

Reply:
correct, thanks, will fix